### PR TITLE
arm: dts: Cleanup HAS_DTS

### PR DESCRIPTION
**Pull request by galak**
Now that all ARM platforms have a device tree we can move selecting of
HAS_DTS up and remove any !HAS_DTS cases, as well as setting in all the
defconfigs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Review by fallrisk — APPROVED** [review]
This looks good to me. I checked all the ARM board directories for HAS_DTS. I built a few of the boards and there generated DTS. I also checked to see that all the ARM boards have the fixup for NUM_IRQ_PRIO_BITS.

File: arch/arm/core/cortex_m/Kconfig [hunk 1]
```diff
@@ -129,15 +129,6 @@ config STACK_ALIGN_DOUBLE_WORD
 	This is needed to conform to AAPCS, the procedure call standard for
 	the ARM. It wastes stack space.
 
-config NUM_IRQ_PRIO_BITS
```
Comment (MaureenHelm) [inline arch/arm/core/cortex_m/Kconfig]: Is there a reason this didn't depend on !HAS_DTS like the others?

**Review by MaureenHelm — COMMENTED** [review]
There are a few other spots that HAS_DTS remains. Do you plan to clean those up too?

File: arch/arm/core/cortex_m/Kconfig (hunk 1 above)
Comment (galak) [inline arch/arm/core/cortex_m/Kconfig]: Not sure why it didn't depend on !HAS_DTS.  The other few places I'll cleanup with more specific SoC related patches.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**